inc/functions_search.php

```diff
@@ -150,6 +150,71 @@ function get_unsearchable_forums($pid=0, $first=1)
 	return $unsearchable;
 }
 
+/**
+ * Build query condition for threads/posts the user is allowed to see.
+ * Will return for example:
+ *  - visibile = 1 - for normal users
+ *  - visible >=1 - for admins & super mods
+ *  - (visibile = 1 OR (visibile = ? AND fid IN ...)) - for forum moderators
+ * 
+ * @param string $table_alias The alias of the table eg t to use t.visible instead of visible
+ * @return string the query condition 
+ */
+function get_visible_where($table_alias = null)
```

Comment:
Maybe we could add another parameter here for specifying forum IDs?

```suggestion
 * @param array $fids The IDs of forums to be searched in. An empty array means all forums.
 * @return string the query condition 
 */
function get_visible_where($table_alias = null, $fids = array())
```

Reply:
I think best to not add extra complexity by passing fids, best to assume all forums to avoid possible bugs where some fids might not be passed by accident.